lib/coin_flip_demo.dart

```diff
@@ -123,6 +121,23 @@ class _Coin extends StatelessWidget {
   }
 }
 
+class _CoinPainter extends CustomPainter {
+  @override
+  void paint(Canvas canvas, Size size) {
+    Paint p = Paint()
+      ..color = Colors.amber;
+
+    Offset center = Offset(size.width / 2, size.height / 2);
+    double radius = min(size.width / 2, size.height / 2);
+    canvas.drawCircle(center, radius, p);
+  }
+
+  @override
+  bool shouldRepaint(CustomPainter oldDelegate) {
```

Comment:
this should return false.
@override
bool shouldRepaint(_CoinPainter oldDelgate) => false;

I think the convention to use a covariant override here as that communicates the intent a bit more clearly.